Src/FluentAssertions/StringWithOccurenceConstraint.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Diagnostics;
+using FluentAssertions.Execution;
+using FluentAssertions.Primitives;
+
+namespace FluentAssertions
+{
+    [DebuggerNonUserCode]
+    public class StringWithOccurenceConstraint
+    {
+        public delegate bool StringOccurrenceComparer(int occuredTimes, int expectedTimes);
+
+        private readonly StringAssertions assertion;
+        private readonly StringOccurrenceComparer comparator;
+        private readonly string expected;
+        private readonly string mode;
+        private readonly int occurrences;
+
+        public StringWithOccurenceConstraint(StringAssertions parentConstraint,
+            string expected,
+            int occurrences,
+            string mode,
+            StringOccurrenceComparer comparator)
```

Comment:
❓ Why a `delegate` instead of `Func<int, int, bool>`?

Reply:
Personally, I think that if you have more than one parameter of the same type then delegate is more readable.